tests/Doctrine/Tests/DBAL/Schema/Visitor/AddDefaultNamespaceTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Schema\Visitor;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\PostgreSqlPlatform;
+use Doctrine\DBAL\Platforms\SQLServerPlatform;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\SchemaConfig;
+use Doctrine\DBAL\Schema\Visitor\AddDefaultNamespace;
+
+/**
+ * @group DBAL-1168
+ */
+class AddDefaultNamespaceTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider namespacedPlatforms
+     */
+    public function testNoNamespacesAndDefaultNamespaceAdded(AbstractPlatform $platform)
+    {
+        $config = new SchemaConfig();
+        $config->setName('test');
+        $schema = new Schema(array(), array(), $config);
+        $schema->createTable('foo');
+
+        $this->assertEmpty($schema->getNamespaces());
+
+        $schema->visit(new AddDefaultNamespace($platform));
+
+        $this->assertNotEmpty($schema->getNamespaces());
+        $this->assertCount(1, $schema->getNamespaces());
+    }
+
+    /**
+     * @dataProvider namespacedPlatforms
+     */
+    public function testCustomNamespaceAndDefaultNamespaceAdded(AbstractPlatform $platform)
+    {
+        $config = new SchemaConfig();
+        $config->setName('test');
+        $schema = new Schema(array(), array(), $config);
+        $schema->createTable('foo');
+        $schema->createTable('bar.baz');
+
+        $this->assertNotEmpty($schema->getNamespaces());
+        $this->assertCount(1, $schema->getNamespaces());
+
+        $schema->visit(new AddDefaultNamespace($platform));
+
+        $this->assertNotEmpty($schema->getNamespaces());
+        $this->assertCount(2, $schema->getNamespaces());
+    }
+
+    /**
+     * @dataProvider namespacedPlatforms
+     */
+    public function testExplicitDefaultNamespaceAndDefaultNamespaceNotAdded(AbstractPlatform $platform)
+    {
+        $config = new SchemaConfig();
+        $config->setName('test');
+        $schema = new Schema(array(), array(), $config);
+        $schema->createTable('foo');
+        $schema->createTable($platform->getDefaultSchemaName().'.baz');
+
+        $this->assertNotEmpty($schema->getNamespaces());
+        $this->assertCount(1, $schema->getNamespaces());
+
+        $schema->visit(new AddDefaultNamespace($platform));
+
+        $this->assertNotEmpty($schema->getNamespaces());
+        $this->assertCount(1, $schema->getNamespaces());
+    }
+
+    public function namespacedPlatforms()
+    {
+        return [
+            [new PostgreSqlPlatform()],
+            [new SQLServerPlatform()],
```

Comment:
It will not work there as it should be added only if platform support default namespace. You think it should be added in ORM without such check and do this check inside itself?